src/gpu/nufft_synthesis.cpp

```diff
@@ -42,8 +42,13 @@ NufftSynthesis<T>::NufftSynthesis(std::shared_ptr<ContextInternal> ctx, NufftSyn
       [&](auto&& arg) -> void {
         using ArgType = std::decay_t<decltype(arg)>;
         if constexpr (std::is_same_v<ArgType, Partition::Grid>) {
+          ctx_->logger().log(BIPP_LOG_LEVEL_INFO, "image partition: grid ({}, {}, {})",
```

Comment:
Why is the call to the logger so different here? would it be possible to harmonize the ctx.logger() vs ctx_->logger() calls across all the files, if only to make searching easier?

Reply:
`ctx_` is a shared pointer, since this is a private class member that has shared ownershop of the context. In other functions, there is no ownership transfer involved, so the context is a reference instead. In C++, `->` is used to access a member of the object the pointer is pointing to.